docs: Add more details for the Cluster Mesh key rotation (#28145)
cilium-team/cilium: Documentation/security/network/encryption-ipsec.rst

from 15 to 1. The Cilium agent will default to ``KEYID`` of zero if its not
specified in the secret.

If you are using Cluster Mesh, the key rotation procedure needs to be applied

> to all clusters in the mesh. You might need to adjust the transition time to

Reviewer: How about making it more specific, `adjust` -> `increase`?

Author: Done, thanks.

> If you are using Cluster Mesh, the key rotation procedure needs to be applied
> to all clusters in the mesh. You might need to increase the transition time to
> allow for the new keys to get deployed and applied across all clusters,
> you can do that with the agent flag ``ipsec-key-rotation-duration``.

Reviewer 2: Minor edits for clarity
```suggestion
If you are using Cluster Mesh, you must apply the key rotation procedure
to all clusters in the mesh. You might need to increase the transition time to
allow for new keys to be deployed and applied across all clusters,
which you can do with the agent flag ``ipsec-key-rotation-duration``.
```

Author: Applied, thanks.
